src/language/concurrency.md

```diff
@@ -237,303 +287,272 @@ After handling the events, the isolate exits.
+[`Isolate.spawnUri()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/spawnUri.html
 
-Check out the [long_running_isolate.dart][] sample,
-which shows how to spawn a long-running isolate
-that receives and sends messages multiple times between isolates.
+### Limitations of isolates
 
-{% assign samples = "https://github.com/dart-lang/samples/tree/main/isolates" %}
+#### Isolates aren't threads
 
-[isolate samples]: {{ samples }}
-[send_and_receive.dart]: {{ samples }}/bin/send_and_receive.dart
-[long_running_isolate.dart]: {{ samples }}/bin/long_running_isolate.dart
+If you’re coming to Dart from a language with multithreading, it’d be reasonable
+to expect isolates to behave like threads, but that isn’t the case. Each isolate
+has its own memory heap, ensuring that none of the state in an isolate is
+accessible from any other isolate. Therefore, isolates are limited by their
+access to their own memory. 
 
+For example, if you have an application with a
+global mutable variable, that variable will be a separate
+variable in your spawned isolate. If you mutate that variable in the spawned
+isolate, it will remain untouched in the main isolate. This is how isolates are
+meant to function, and it's important to keep in mind when you’re considering
+using isolates.
 
-## Performance and isolate groups
+#### Message types
 
-When an isolate calls [`Isolate.spawn()`][],
-the two isolates have the same executable code
-and are in the same _isolate group_.
-Isolate groups enable performance optimizations such as sharing code;
-a new isolate immediately runs the code owned by the isolate group.
-Also, `Isolate.exit()` works only when the isolates
-are in the same isolate group.
+Messages sent via [`SendPort`][]
+can be almost any type of Dart object, but there are a few exceptions:
 
-In some special cases,
-you might need to use [`Isolate.spawnUri()`][],
-which sets up the new isolate with a copy of the code
-that's at the specified URI.
-However, `spawnUri()` is much slower than `spawn()`,
-and the new isolate isn't in its spawner's isolate group.
-Another performance consequence is that message passing
-is slower when isolates are in different groups.
+- Objects with native resources, such as [`Socket`][].
+- [`ReceivePort`][]
+- [`DynamicLibrary`][]
+- [`Finalizable`][]
+- [`Finalizer`][]
+- [`NativeFinalizer`][]
+- [`Pointer`][]
+- [`UserTag`][]
+- Instances of classes that either themselves are marked with @pragma
+  ('vm:isolate-unsendable'), extend or implement such classes cannot be sent
+  through the ports.
 
-[`Isolate.spawnUri()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/spawnUri.html
+Apart from those exceptions, any object can be sent. 
+Check out the [`SendPort`][] documentation for more information.
```

Comment:
Consider linking directly to `SendPort.send` docs here since almost all of the details are there.